### ISSUE TITLE
Feature: Check if all nodes, in current state, are reachable before applying new changes

### ISSUE THREAD

**Issue by Despire** [berops/claudie#1516]
### Motivation
If some of the nodes are not reachable before we apply changes to an already existing current state at some point during the workflow the new changes will fail. To prevent this issue and save time we should check if nodes in the current state are reachable first.

### Description

Before a new Task is being scheduled https://github.com/berops/claudie/blob/844eb7afe50aaa8241e27b47a234875a1ee9fb24/services/manager/internal/service/handler_next_task.go#L31, check if current state exists and if yes check if all the nodes are reachable.

### Exit criteria

- [ ] Check if all nodes in current state are reachable before making changes.

**Comment by bernardhalas** [comment]
Makes sense. Assess pros and cons for placing this in builder and ansibler.

**Comment by Despire** [comment]
This has been implemented by the unreachable nodes check. https://github.com/berops/claudie/pull/1658
